api/workers/events.go

```diff
@@ -0,0 +1,298 @@
+
+		req.LogEvents = []*cloudwatchlogs.InputLogEvent{
+			&cloudwatchlogs.InputLogEvent{
+				Message:   aws.String(log),
+				Timestamp: aws.Int64(time.Now().UnixNano() / 1000000),
+			},
+		}
+
+		pres, err := models.CloudWatchLogs().PutLogEvents(req)
+		if err != nil {
+			return err
+		}
+
+		stream.SequenceToken = *pres.NextSequenceToken
+		appLogStreams[app] = stream
+
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func handleAccountEvents() {
+	err := processQueue("AccountEvents", func(body string) error {
+		var e event
+
+		if err := json.Unmarshal([]byte(body), &e); err != nil {
+			return err
+		}
+
+		switch e.DetailType {
+		case "ECS Task State Change":
+			var detail detailTaskStateChange
+
+			if err := remarshal(e.Detail, &detail); err != nil {
+				return err
+			}
+
+			if !strings.HasPrefix(detail.Group, "service:") {
+				return nil
+			}
+
+			parts := strings.Split(detail.ClusterArn, "/")
+			cluster := parts[len(parts)-1]
+			service := strings.TrimPrefix(detail.Group, "service:")
+			app := strings.Split(strings.TrimPrefix(service, fmt.Sprintf("%s-", os.Getenv("RACK"))), "-Service")[0]
+
+			if detail.LastStatus == "PENDING" {
+				res, err := models.ECS().DescribeServices(&ecs.DescribeServicesInput{
+					Cluster:  aws.String(cluster),
+					Services: []*string{aws.String(service)},
+				})
+				if err != nil {
+					return err
+				}
+
+				if len(res.Services) < 1 {
+					return fmt.Errorf("could not find service: %s", service)
+				}
+
+				stream, err := getAppLogStream(app)
+				if err != nil {
+					return err
+				}
+
+				events := res.Services[0].Events
+
+				if len(events) > 5 {
+					events = events[0:5]
+				}
+
+				req := &cloudwatchlogs.PutLogEventsInput{
+					LogGroupName:  aws.String(stream.LogGroup),
+					LogStreamName: aws.String(stream.LogStream),
+				}
+
+				if stream.SequenceToken != "" {
+					req.SequenceToken = aws.String(stream.SequenceToken)
+				}
+
+				for _, e := range events {
+					req.LogEvents = append(req.LogEvents, &cloudwatchlogs.InputLogEvent{
+						Message:   aws.String(fmt.Sprintf("AWS/ECS %s", *e.Message)),
+						Timestamp: aws.Int64(e.CreatedAt.UnixNano() / 1000000),
+					})
+				}
+
+				// havent done this in a while
+				// TODO use sort.Slice once we upgrade to 1.8
+				for i := 0; i < len(req.LogEvents)-1; i++ {
+					for j := i + 1; j < len(req.LogEvents); j++ {
+						if *req.LogEvents[i].Timestamp > *req.LogEvents[j].Timestamp {
+							req.LogEvents[i], req.LogEvents[j] = req.LogEvents[j], req.LogEvents[i]
+						}
+					}
+				}
+
+				pres, err := models.CloudWatchLogs().PutLogEvents(req)
+				if err != nil {
+					return err
+				}
+
+				stream.SequenceToken = *pres.NextSequenceToken
+				appLogStreams[app] = stream
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func getAppLogStream(app string) (appLogStream, error) {
```

Comment:
this name is a bit confusing since this gets and/or creates a log group for an app or a rack. Maybe getOrCreateStackLogStream?